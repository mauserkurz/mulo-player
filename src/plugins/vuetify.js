import Vue from 'vue';
import Vuetify from 'vuetify';
import { THEME as theme } from '@/constants';

Vue.use(Vuetify);

export default new Vuetify({ theme });
