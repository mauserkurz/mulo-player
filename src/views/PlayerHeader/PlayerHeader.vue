<template>
  <v-row
    class="player-header"
    align="center"
    justify="space-between">
    <a href="/player" title="Music lover">
      <img alt="App logo" src="@/assets/logo.svg" width="64" height="64">
    </a>

    <v-btn
      color="primary"
      outlined
      @click="logout">
      Logout
    </v-btn>
  </v-row>
</template>

<script>
export default {
  name: 'PlayerHeader',

  methods: {
    logout() {
      this.$emit('logout');
    },
  },
};
</script>

<style scoped lang="less">
.player-header {}
</style>
