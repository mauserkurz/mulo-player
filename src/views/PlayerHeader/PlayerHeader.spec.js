import { mount, shallowMount, createLocalVue } from '@vue/test-utils';
import vuetify from '@/plugins/vuetify';
import PlayerHeader from './PlayerHeader.vue';

const localVue = createLocalVue();

describe('Component PlayerHeader', () => {
  const createWrapper = ({ isWithoutStubs = false, ...options } = {}) => {
    const renderer = isWithoutStubs ? mount : shallowMount;
    const wrapper = renderer(PlayerHeader, {
      localVue,
      vuetify,
      ...options,
    });

    return { wrapper };
  };

  describe('snapshots', () => {
    it('should match snapshot', () => {
      const { wrapper } = createWrapper({ isWithoutStubs: true });

      expect(wrapper.html()).toMatchSnapshot();
    });
  });

  describe('events', () => {
    it('should emit logout event after click on logout button', () => {
      const { wrapper } = createWrapper({ isWithoutStubs: true });

      wrapper.find('.v-btn').trigger('click');
      expect(wrapper.emitted().logout).toEqual([[]]);
    });
  });
});
