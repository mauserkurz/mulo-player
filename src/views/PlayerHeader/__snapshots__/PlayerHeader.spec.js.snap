// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component PlayerHeader snapshots should match snapshot 1`] = `
<div class="row player-header align-center justify-space-between"><a href="/player" title="Music lover"><img alt="App logo" src="@/assets/logo.svg" width="64" height="64"></a> <button type="button" class="v-btn v-btn--depressed v-btn--flat v-btn--outlined theme--light v-size--default primary--text"><span class="v-btn__content">
    Logout
  </span></button></div>
`;
